jupyter_book/cli/main.py

```diff
@@ -216,28 +227,29 @@ def build(
         BUILD_PATH = Path(BUILD_PATH).joinpath("_build")
 
         # Table of contents
-        if toc is None:
-            toc = PATH_SRC_FOLDER.joinpath("_toc.yml")
-        else:
-            toc = Path(toc)
-
-        if not toc.exists():
-            _error(
-                "Couldn't find a Table of Contents file. To auto-generate "
-                f"one, run\n\n\tjupyter-book toc {path_source}"
-            )
+        toc = PATH_SRC_FOLDER.joinpath("_toc.yml") if toc is None else Path(toc)
 
-        # Check whether the table of contents has changed. If so we rebuild all
-        build_files = list(BUILD_PATH.joinpath(".doctrees").rglob("*"))
-        if toc and build_files:
-            toc_modified = toc.stat().st_mtime
+        if not get_config_only:
 
-            build_modified = max([os.stat(ii).st_mtime for ii in build_files])
+            if not toc.exists():
+                _error(
+                    "Couldn't find a Table of Contents file. "
+                    "To auto-generate one, run:"
+                    f"\n\n\tjupyter-book toc from-site {path_source}"
+                )
 
-            # If the toc file has been modified after the build we need to force rebuild
-            freshenv = toc_modified > build_modified
+            # we don't need to read the toc here, but do so to control the error message
+            try:
+                parse_toc_yaml(toc)
+            except MalformedError as exc:
+                _error(
+                    f"The Table of Contents file is malformed: {exc}\n"
+                    "You may need to migrate from the old format, using:"
+                    f"\n\n\tjupyter-book toc migrate {toc}"
```

Comment:
```suggestion
                    f"\n\n\tjupyter-book toc migrate {toc} -o {toc}"
```

Perhaps we could add an `-o` flag to write the output somewhere, and then raise this error message, per @mmcky's comment 1?

Reply:
done